src/rust/engine/fs/store/src/lib.rs

```diff
@@ -806,15 +806,16 @@ impl Store {
               remote_store
                 .clone()
                 .maybe_upload(digest, async move {
-                  // TODO(John Sirois): Consider allowing configuration of when to buffer large blobs
-                  // to disk to be independent of the remote store wire chunk size.
-                  if digest.size_bytes > remote_store.store.chunk_size_bytes() {
-                    Self::store_large_blob_remote(local, remote_store.store, entry_type, digest)
-                      .await
-                  } else {
-                    Self::store_small_blob_remote(local, remote_store.store, entry_type, digest)
-                      .await
-                  }
+                  match local.load_from_fs(digest).await? {
+                    Some(path) => {
+                      Self::store_fsdb_blob_remote(remote_store.store, digest, path).await?
+                    }
+                    None => {
+                      Self::store_lmdb_blob_remote(local, remote_store.store, entry_type, digest)
```

Comment:
As the comment in `store_lmdb_blob_remote` now notes, this code makes the assumption that anything in LMDB is small enough to load into memory. AIUI, `LARGE_FILE_SIZE_LIMIT` is 512KB, i.e. these blobs should be that large at most. 

Do you think that's reasonable?

Reply:
Yea, I do think that that is reasonable.